src/Orleans/CodeGeneration/GrainInterfaceUtils.cs

```diff
@@ -126,6 +126,9 @@ public static bool IsStatelessWorker(MethodInfo methodInfo)
         
         public static int ComputeMethodId(MethodInfo methodInfo)
         {
+            var attr = methodInfo.GetCustomAttribute<MethodIdAttribute>(true);
+            if (attr != null) return attr.MethodId;
```

Comment:
Compilation will fail because we generate a switch case in the generated `GrainReference`.

Maybe we could add simple check in `CodeGeneratorCommon.GenerateGrainInterfaceAndMethodSwitch`, but I don't know how to surface the error, but I think we can merge as is.